lib/puppet/configurer.rb

```diff
@@ -395,7 +395,8 @@ def find_functional_server
       host = server[0]
       port = server[1] || Puppet[:masterport]
       begin
-        http = Puppet::Network::HttpPool.http_ssl_instance(host, port)
+        ssl_context = Puppet.lookup(:ssl_context)
+        http = Puppet::Network::HttpPool.connection(host, port.to_i, ssl_context: ssl_context)
```

Comment:
@jhelwig suggesting to make `Puppet.lookup(:ssl_context)` the default value for the `ssl_context` keyword argument.

Reply:
I changed this so callers don't have to pass in a value and will default to the global one.

Reply:
After thinking about this more, I view the `HttpPool.connection` method as the API for the http client subsystem. I don't want its implementation to rely on global state. It is inconvenient for callers to have to lookup the context, but then it's explicit, and we avoid the problem where the caller forgets to pass the context and ends up using the wrong one. Right now that problem can't really happen, since there's only one context. But very shortly we'll need to make connections with different contexts as we bootstrap SSL.